src/assisted_test_infra/test_infra/helper_classes/cluster.py

```diff
@@ -801,10 +803,46 @@ def download_image(self, iso_download_path: str = None) -> Path:
             )
         return self._infra_env.download_image(iso_download_path)
 
+    # Downloads the iPXE script from assisted service to a local file
+    def prepare_local_ipxe_server(self, infra_env_id):
+        assisted_service_url = utils.get_assisted_service_url_by_configmap("assisted-service-config", "assisted-installer")
+        remote_ipxe_path = f"/api/assisted-install/v2/infra-envs/{infra_env_id}/downloads/files?file_name=ipxe-script"
+        ipxe_download_url = f"{assisted_service_url}{remote_ipxe_path}"
+
+        local_ipxe_script_file = f"{self._config.local_ipxe_server_artifacts_path}/{self._config.cluster_name}"
+
+        if not os.path.exists(local_ipxe_script_file):
+            log.info("Local iPXE server path doesn't exist, creating the folder: %s", local_ipxe_script_file)
+            utils.recreate_folder(os.path.dirname(local_ipxe_script_file), force_recreate=False)
+        utils.download_file(ipxe_download_url, local_ipxe_script_file, False)
+
+    # Runs a local python server that hosts the iPXE scripts 
+    def run_local_ipxe_server(self):
+        ip = re.search('\d+\.\d+\.\d+\.\d+',self._config.local_ipxe_server_url).group(0)
+        log.info("the ip %s", ip)
+        run_ipxe_server_cmd = f'''python3 {self._config.local_ipxe_server_script_path} \
+            --dir {self._config.local_ipxe_server_artifacts_path} \
+            --ip {ip} \
+            --port {self._config.local_ipxe_server_port}'''
+        log.info("the cmd %s", run_ipxe_server_cmd)
+        utils.run_command(run_ipxe_server_cmd, run_in_background=True)
+
     @JunitTestCase()
     def prepare_for_installation(self, **kwargs):
-        super(Cluster, self).prepare_for_installation(**kwargs)
+        if self._config.ipxe_boot:
+            # The infra-env needs to be generated here since we're not downloading the ISO locally
+            infra_env = self.generate_infra_env()
+            
+            self.prepare_local_ipxe_server(infra_env.id)
+            self.run_local_ipxe_server()
+
+            local_ipxe_server_url = f"{self._config.local_ipxe_server_url}:{self._config.local_ipxe_server_port}/{self._config.cluster_name}"
+            network_name = self.nodes.get_cluster_network()
+            
+            libvirt_controller = LibvirtController(config=TerraformConfig(), entity_config=self._config)
+            libvirt_controller.set_ipxe_url(network_name=network_name, ipxe_url=local_ipxe_server_url)
```

Comment:
Also here, create an ipxe controller as I mentioned before, and put create a method from this code.
Note that we are creating a controller only in our test preparation (`tests/base_test.py` fixtures), I think that this method needs to run as a fixture before test starts (maybe after cluster creation)

Reply:
Created the controller!